gsoc_final_post/inigo_tellaetxe_gsoc_2024_final_report.rst

```diff
@@ -0,0 +1,226 @@
+  * Training was run for 120 epochs, using a data set containing plausible and implausible streamlines. All the training experiments in my GSoC work were done with this data set. The figure below shows a summary of the replication results, which consist of running a set of unseen plausible streamlines through the model (encoder and decoder):
+
+    .. image:: https://github.com/dipy/dipy.org/blob/master/_static/images/gsoc/2024/inigo/fibercup_replicated.png
+      :alt: Replication of the FINTA architecture results on the FiberCup dataset.
+      :align: center
+      :width: 800
+  
+    We see that the Keras result is very similar to the PyTorch result, though the fiber crossings are not as well resolved in the Keras result. This is not a thing to worry since this is not systematic, as seen in other sets of unseen data.
+
+
+* **Implemented a Variational AutoEncoder (VAE) architecture based on the** `FINTA <https://doi.org/10.1016/j.media.2021.102126>`_ **AE architecture.**
+
+  * After thorough discussion with my mentors, we decided that regularizing the latent space could be a good idea due to two reasons: to avoid "white spaces" in the latent space it that could lead undefined points due to the possible discontinuity of the space; and to prevent the model from learning the identity function, what is a known issue with unregularized AutoEncoders, though it does not always happen. The only addition to the original AE architecture was the inclusion of the Kullback-Leibler divergence loss term in the loss function, and two dense layers after the convolutional layers of the encoder, to output the ::math:`\mu` and ::math:`\sigma^2` of the latent space vector ::math:`z`.
+
+  * The model is found in the ``vae_model.py`` module of the `TractoEncoder GSoC <https://github.com/itellaetxe/tractoencoder_gsoc>`_ repository. The architecture can be summarized in the figure below:
+
+  .. image:: https://github.com/dipy/dipy.org/blob/master/_static/images/gsoc/2024/inigo/inigo_variational_autoencoder.png
+    :alt: VAE architecture diagram.
+    :align: center
+    :width: 800
+
+  * It was necessary to implement Batch Normalization after the Encoder convolutional layers and exponential operator clipping, to prevent gradient explosion while training the model. The figure above displays that the encoder outputs the variance of the latent space vector (::math:`\sigma^2`), though it is a common practice to output the log-variance instead. This leads to using ::math:`\sigma = e^{\frac{1}{2}\text{log}{\sigma^2}}`, which can get numerically quite unstable as it can shoot up very quickly when a sufficiently big log-variance is input. This modification allowed stable training and it was a major contribution to the robustness of the architecture. Shoutout to my lab colleague `Jorge <https://github.com/jgarciacondado>`_ for the ideas and discussions on the topic.
```

Comment:
Some errors in markdown? When I looked at the compiled file it said \sigma instead of $\\sigma$

Reply:
Remove double colon substitute for single colon -> :math: